public/assets/js/index.js

```diff
@@ -0,0 +1,220 @@
+const loginForm = $("#login-form");
+const signupForm = $("#sign-up-form");
+const newBlogForm = $("#new-blog-form");
+const commentForm = $("#comment-form");
+const editBlogForm = $("#edit-blog-form");
+const logoutBtn = $("#logout-btn");
+const deleteYesBtn = $("#delete-yes-btn");
+const errorMessageDiv = $("#error-message");
+const confirmDeleteModal = $("#confirm-delete-modal");
+
+const renderError = (message) => {
+  errorMessageDiv.empty();
+
+  const errorDiv = `<div class="form-text text-danger"><span class="mx-2">${message}</span><i class="fas fa-exclamation-circle"></i></div>`;
+
+  errorMessageDiv.append(errorDiv);
+};
+
+const makeRequest = async (url, method, body = {}) => {
+  const response = await fetch(url, {
+    method,
+    headers: {
+      "Content-type": "application/json",
+    },
+    body: JSON.stringify(body),
+  });
+
+  const data = await response.json();
+
+  return data;
+};
+
+const handleLogin = async (event) => {
+  event.preventDefault();
+
+  const email = $("#email").val();
+  const password = $("#password").val();
+
+  if (email && password) {
+    try {
+      const data = await makeRequest("/auth/login", "POST", {
+        email,
+        password,
+      });
+
+      if (data.success) {
+        window.location.replace("/dashboard");
+      } else {
+        renderError("Failed to login");
```

Comment:
You could amend these messages slightly to reflect why the login failed e.g. both fields were not entered etc. This will give the user a better idea of what they are doing wrong.